lib/unfurls/index.js

```diff
@@ -1,105 +1,88 @@
+    .map(async (link) => {
+      const eligible = await Unfurl.isEligibleForUnfurl(req.body.team_id, event.channel, link.url);
+      if (eligible) {
+        req.log.debug(link, 'Link eligible for unfurls');
+        return link;
+      }
+      req.log.debug(link, 'Link not eligible for unfurl');
+      return null;
+    }));
+  eligibleLinks = eligibleLinks.filter(link => link);
 
-  // @todo: Temporary workaround for unfurling public links
-  const github = new GitHubApi({ logger: req.log });
-  if (process.env.GITHUB_TOKEN) {
-    github.authenticate({
-      type: 'token',
-      token: process.env.GITHUB_TOKEN,
-    });
+  if (eligibleLinks.length === 0) {
+    return res.send();
   }
 
-  const unfurls = {};
-
-  await Promise.all(event.links.map(async (link) => {
-    if (await isEligibleForUnfurl(req.body.team_id, event.channel, link.url)) {
-      req.log.debug(link, 'Link eligible for unfurls');
-      try {
-        unfurls[link.url] = await unfurl(github, link.url, type);
-      } catch (err) {
-        if (err instanceof UnsupportedResource || err.code === 404) {
-          req.log.debug(err, 'Could not unfurl');
-        } else {
-          throw err;
-        }
+  // if there's 1 link in the message, full unfurl
+  // if there are 2 links in the message, condensed unfurl for both
+  const type = eligibleLinks.length === 1 ? 'full' : 'condensed';
+
+  const workspace = await SlackWorkspace.findOne({
+    where: { slackId: req.body.team_id },
+  });
+
+  await Promise.all(eligibleLinks.map(async (link) => {
+    let newUnfurl;
+    try {
+      newUnfurl = await Unfurl.spawn({
+        teamId: req.body.team_id,
+        userId: event.user,
+        channel: event.channel,
+        url: link.url,
+        isCondensed: type === 'condensed',
+        slackMessageTimestamp: event.message_ts,
+      });
+    } catch (err) {
+      if (err.name === 'UnsupportedResource' || err.code === 404) {
+        req.log.debug(err, 'Could not get unfurl attachment');
+        return Promise.resolve();
+      }
+      throw err;
+    }
+    if (process.env.ALLOWED_CHANNELS && process.env.ALLOWED_CHANNELS.split(',').includes(event.channel)) {
+      if (newUnfurl && !newUnfurl.isDelivered) {
+        await workspace.client.chat.postEphemeral({
+          channel: event.channel,
+          user: event.user,
+          attachments: [new PrivateUnfurlPrompt(newUnfurl).getAttachment()],
+        });
       }
-    } else {
-      req.log.debug(link, 'Link not eligible for unfurl');
     }
   }));
+  return res.send();
+}
 
-  if (Object.keys(unfurls).length !== 0) {
-    const workspace = await SlackWorkspace.findOne({
-      where: { slackId: req.body.team_id },
-    });
-
-    req.log.debug(unfurls, 'Unfurling links');
-    const slackRes = await workspace.client.chat.unfurl({
-      ts: event.message_ts,
-      channel: event.channel,
-      unfurls,
+async function unfurlAction(req, res) {
+  if (req.body.actions[0].name === 'unfurl-dismiss') {
```

Comment:
Question: When the unfurl is dismissed, should we delete the db record? Or should we have another `isDismissed` column that we set to `true` when it gets dismissed?

Reply:
Is there ay reason to keep it around? (Besides analytics, which we discussed using a different tool for those purposes)

Reply:
No reason to keep it around besides analytics and building up an understanding of in which cases folks decide not to unfurl. Although this current implementation is doesn't fully serve the analytics use case. To fully serve that use case we'd need `isDismissed`, which is probably not a good thing to implement at this stage. 

I'll add some code to implement the deleting logic.